samples/subsys/ipc/openamp_rsc_table/src/main_remote.c

```diff
@@ -243,14 +270,72 @@ platform_create_rpmsg_vdev(unsigned int vdev_index,
 	return NULL;
 }
 
-void app_task(void *arg1, void *arg2, void *arg3)
+void app_rpmsg_client_sample(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+	unsigned int msg_cnt = 0;
+	int ret = 0;
+
+	k_sem_take(&data_sc_sem,  K_FOREVER);
+
+	printk("\r\nOpenAMP[remote] Linux sample client responder started\r\n");
+
+	ret = rpmsg_create_ept(&sc_ept, rpdev, "rpmsg-client-sample",
+			       RPMSG_ADDR_ANY, RPMSG_ADDR_ANY,
+			       rpmsg_recv_cs_callback, NULL);
+
+	while (msg_cnt < 100) {
+		k_sem_take(&data_sc_sem,  K_FOREVER);
+		msg_cnt++;
+		rpmsg_send(&sc_ept, sc_msg.data, sc_msg.len);
+	}
+	rpmsg_destroy_ept(&sc_ept);
+
+	printk("OpenAMP Linux sample client responder ended\n");
+}
+
+void app_rpmsg_tty(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+	unsigned char tx_buff[512];
+	int ret = 0;
+
+	k_sem_take(&data_tty_sem,  K_FOREVER);
+
+	printk("\r\nOpenAMP[remote] Linux tty responder started\r\n");
+
+	tty_ept.priv = &tty_msg;
+	ret = rpmsg_create_ept(&tty_ept, rpdev, "rpmsg-tty",
+			       RPMSG_ADDR_ANY, RPMSG_ADDR_ANY,
+			       rpmsg_recv_tty_callback, NULL);
+
+	while (tty_ept.addr !=  RPMSG_ADDR_ANY) {
+		k_sem_take(&data_tty_sem,  K_FOREVER);
+		if (tty_msg.len) {
+			snprintf(tx_buff, 13, "TTY 0x%04x: ", tty_ept.addr);
```

Comment:
The compiler warns for `implicit declaration of function 'snprintf'`, and suggests to add `#include <stdio.h>`.